orderer/common/cluster/connectionsmgr.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cluster
+
+import (
+	"sync"
+
+	"github.com/hyperledger/fabric/internal/pkg/comm"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+type ConnectionsCache map[string]*grpc.ClientConn
+
+func (cbc ConnectionsCache) Lookup(key string) (*grpc.ClientConn, bool) {
+	conn, ok := cbc[key]
+	return conn, ok
+}
+
+func (cbc ConnectionsCache) Put(key string, conn *grpc.ClientConn) {
+	cbc[key] = conn
+}
+
+func (cbc ConnectionsCache) Remove(key string) {
+	delete(cbc, key)
+}
+
+func (cbc ConnectionsCache) Size() int {
+	return len(cbc)
+}
+
+type ConnectionsMgr struct {
+	lock        sync.RWMutex
+	Connections ConnectionsCache
+	dialer      comm.ClientConfig
+}
+
+func (c *ConnectionsMgr) Connect(endpoint string, serverRootCACert []byte) (*grpc.ClientConn, error) {
+	if serverRootCACert == nil {
+		return nil, errors.New("Server Root CA Cert is nil")
+	}
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	conn, alreadyConnected := c.Connections.Lookup(endpoint)
+	if alreadyConnected {
+		return conn, nil
+	}
+
+	c.dialer.SecOpts.ServerRootCAs = [][]byte{serverRootCACert}
```

Comment:
I am not sure this is good practice, to change the configuration of a shared field that is accessed via different goroutines. 

Can't we clone it first and then override the server Root CA certs? 
Other goroutines may use this reference later without a lock, we should not assume anything about how the field `ServerRootCAs` is used later on.

Reply:
agree. i will update it